File: api/src/endpoint/user/has_user.rs
use serde::{Deserialize, Serialize};
use api_derive::EndpointRequest;

use crate::endpoint::Endpoint;

pub struct HasUser;

impl Endpoint for HasUser {
    const PATH: &'static str = "/user/{user_name}";
    const METHOD: http::Method = http::Method::HEAD;

    type Request = HasUserApiRequest;
    type Response = HasUserResponse;
}

#[derive(EndpointRequest, Debug, Serialize, Deserialize)]
pub struct HasUserApiRequest {
    pub user_name: String,
}

pub type HasUserResponse = ();

File: gitdot-api-ts/src/endpoint/user/has-user.ts
import { z } from "zod";

export const HasUserRequest = z.object({
  user_name: z.string(),
});
export type HasUserRequest = z.infer<typeof HasUserRequest>;

export const HasUser = {
  path: "/user/{user_name}",
  method: "HEAD",
  request: HasUserRequest,
  response: z.void(),
} as const;
export type HasUser = typeof HasUser;
